### ISSUE TITLE
[A11Y] Modal window does not trap keyboard focus

### ISSUE THREAD

**Issue by PhilibertDugas** [paypal/paypal-checkout#788]
When the PayPal overlay modal is displayed, users of assistive technology and keyboard-only users are still able to interact with elements "underneath" the window. Keyboard users should be able to trigger the overlay modal and interact with only the modal content until it is dismissed.

**HTML:**

```html
<div id="xcomponent-paypal-checkout-a1bab52533" class="paypal-checkout-context-popup paypal-checkout-overlay paypal-checkout-background-color-black paypal-checkout-logo-color-white">
  <!-- … -->
</div>
```

## Steps to reproduce

1. Go to https://www.chubbiesshorts.com/ with any supported browser
2. Add some items to the cart
3. Proceed to the Cart
4. Activate the PayPal control
5. `Cmd` + `~` back to the main window
6. With an active screen reader, `Tab` or `Shift`+`Tab` through content items

### Results

**Expected:**

* Only content within the modal window to be interacted with

**Actual:**

* Content "behind" the modal window can be interacted with

## Recommendation

Keep keyboard focus within the modal window so users cannot `Tab` or `Shift`+`Tab` away from it without submitting, closing, or pressing the `Esc` key.

## Additional information

* **WCAG Principle:** Operable
* **WCAG SC:** 2.4.3 Focus Order
* **Severity:** High

## Screenshots
<img width="941" alt="screen shot 2018-03-05 at 2 41 22 pm" src="https://user-images.githubusercontent.com/1392632/36995898-605bcece-2083-11e8-983f-6ad7ef314917.png">

**Comment by bluepnume** [comment]
@mcntsh can you take a look? I can point you towards some code we have internally which may help trapping keyboard focus.

**Comment by mcntsh** [comment]
This has been addressed in #896 and will be out in our next release!
